addons/hr_holidays/models/hr_employee.py

```diff
@@ -0,0 +1,106 @@
+            # Find for holidays status
+            status = self.env['hr.holidays.status'].search([('limit', '=', False)])
+            if len(status) != 1:
+                raise UserError(_("The feature behind the field 'Remaining Legal Leaves' can only be used when there is only one leave type with the option 'Allow to Override Limit' unchecked. (%s Found). Otherwise, the update is ambiguous as we cannot decide on which leave type the update has to be done. \nYou may prefer to use the classic menus 'Leave Requests' and 'Allocation Requests' located in 'Human Resources \ Leaves' to manage the leave days of the employees if the configuration does not allow to use this field.") % (len(status)))
+            if not status.id or diff == 0:
+                return False
+            if diff < 0:
+                raise UserError(_('You cannot reduce validated allocation requests'))
+            leave = self.env['hr.holidays'].create({'name': _('Allocation for %s') % self.name, 'employee_id': self.id, 'holiday_status_id': status.id, 'type': 'add', 'holiday_type': 'employee', 'number_of_days_temp': diff})
+            for sig in ('confirm', 'validate', 'second_validate'):
+                leave.signal_workflow(sig)
+            return True
+        return False
+
+    def _compute_remaining_days(self):
+        remaining_leaves = self._remaining_leaves_count()
+        for employee in self:
+            employee.remaining_leaves = remaining_leaves[employee.id]
+
+    @api.depends('current_leave_id', 'leave_date_from', 'leave_date_to')
+    def _compute_leave_status(self):
+        today_date = fields.Date.from_string(fields.Date.today())
+        today_end = fields.Datetime.to_string(today_date + relativedelta(hours=23, minutes=59, seconds=59))
+        holidays = self.env['hr.holidays'].search([
+            ('date_from', '<=', fields.Datetime.now()),
+            ('date_to', '>=', today_end), ('type', '=', 'remove'),
+            ('state', 'not in', ('cancel', 'refuse')),
+            ('employee_id', 'in', self.ids)])
+        for holiday in holidays:
+            employee = holiday.employee_id
+            employee.leave_date_from = holiday.date_from
+            employee.leave_date_to = holiday.date_to
+            employee.current_leave_state = holiday.state
+            employee.current_leave_id = holiday.holiday_status_id.id
+
+    def _compute_leaves_count(self):
+        remaining_leaves = self._remaining_leaves_count()
+        for employee in self.filtered(lambda e: e.id in remaining_leaves):
+            employee.leaves_count = remaining_leaves[employee.id]
+
+    def _compute_show_leaves(self):
+        if self.env.user.has_group('base.group_hr_user'):
+            for employee in self:
+                employee.show_leaves = True
+        else:
+            for employee in self.filtered(lambda e: e.user_id == self.env.user):
+                employee.show_leaves = True
```

Comment:
May be you can optimize the code like:

for employee in self:
    if self.env.user.has_group('base.group_hr_user') or employee.user_id == self.env.user:
    	employee.show_leaves = True